lib/hushed/date_service.rb

```diff
@@ -0,0 +1,55 @@
+module Hushed
+  class DateService
+
+    def build_date(date_string)
+      month = date_string.slice(0..1).to_i
+      day = date_string.slice(2..3).to_i
+      year = ("20" + date_string.slice(4..5)).to_i
+
+      Date.new(year, month, day)
+    end
+
+    def most_recent_day(date_array)
+      quick_sort(date_array).first
+    end
+
+    def quick_sort(date_array)
```

Comment:
It's not that about efficiency (it'll never contain more that a few 10s of items).  Just that we should use whatever features the core library is giving us.  One thing, I had to go through it for a few minutes trying to find why we weren't using Enumerable#sort and if there was anything special about it, so I guess for clarity and to avoid that kind of confusion.  Most of the code in the class is for implementing this sorting algo and non specific to dates.

This sums up pretty well.  It's from Effective Java but this item is language agnostic.  
http://jtechies.blogspot.ca/2012/07/item-47-know-and-use-libraries.html.

Related, you can also use Date.strptime(date_string, '%m%d%y') to build a date from your string.  This class could be rewritten to (no need to create an instance of the class since it contains only utility methods)

```
module Hushed
  module DateService
    extend self

    def build_date(date_string)
      Date.strptime(date_string, '%m%d%y')
    end

    def most_recent_day(date_array)
      date_array.max
    end
  end
end
```